python/cudf/cudf/pandas/_wrappers/numpy.py

```diff
@@ -126,6 +126,23 @@ def ndarray__array_ufunc__(self, ufunc, method, *inputs, **kwargs):
     return result
 
 
+def ndarray__reduce__(self):
+    # As it stands the custom pickling logic used for all other
+    # proxy types is incompatible with our proxy ndarray. The pickle
+    # constructor we use to deserialize the other proxy types calls
+    # object.__new__(type) which you cannot call on subclasses of
+    # numpy arrays because the new array won't be created with numpy's
+    # specific memory management logic. Therefore, we have to handle
+    # serialization separately for proxy arrays.
+    return (
```

Comment:
It would be good to have a dedicated test that round trip pickling of our proxy array returns a proxy array (I don't recall where we put our existing proxy numpy array tests)

Reply:
Good call. We put them in `cudf_pandas_tests/test_cudf_pandas.py`.

Reply:
Added in https://github.com/rapidsai/cudf/pull/17929/commits/82c472e226fc0ec66252b0b21fa55936bc0a6478